src/components/Whiteboard/ElementBehaviors/Resizable/ResizeHandle.tsx

```diff
@@ -124,30 +150,47 @@ export type DragEvent = {
 };
 
 export type ResizeHandleProps = {
-  handlePosition: ResizeHandlePosition;
-  containerWidth: number;
-  containerHeight: number;
   onDrag?: Dispatch<DragEvent>;
   onDragStart?: Dispatch<DragEvent>;
   onDragStop?: Dispatch<DragEvent>;
-};
+} & (LineElementProps | PolylineAndShapeElementsProps);
 
 export function ResizeHandle({
   onDrag,
   onDragStart,
   onDragStop,
-  handlePosition,
-  containerWidth,
-  containerHeight,
+  ...rest
 }: ResizeHandleProps) {
   const nodeRef = useRef<SVGRectElement>(null);
   const { scale, calculateSvgCoords } = useSvgCanvasContext();
-  const { x, y, width, height, cursor } = calculateResizeHandlePosition(
-    handlePosition,
-    containerWidth,
-    containerHeight,
-    scale,
-  );
+
+  let rectAttributesValues;
+
+  if (rest.elementKind === 'line') {
+    const { elementKind, handlePosition, handlePositionX, handlePositionY } =
+      rest;
+
+    rectAttributesValues = calculateResizeHandlePosition({
+      scale,
+      elementKind,
+      handlePosition,
+      handlePositionX,
+      handlePositionY,
+    });
+  } else {
+    const { elementKind, handlePosition, containerWidth, containerHeight } =
+      rest;
+
+    rectAttributesValues = calculateResizeHandlePosition({
+      scale,
+      elementKind,
+      handlePosition,
+      containerWidth,
+      containerHeight,
+    });
+  }
+
+  const { x, y, width, height, cursor } = rectAttributesValues;
```

Comment:
if resize parameters are in a separate prop then this `if` condition is not needed and logic could work for elements of the same kind the same.